[quant][graphmode][fx][fp16] Add fp16 support for sigmoid

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52865 [quant][graphmode][fx][fp16] Add fp16 support for silu
* #52864 [quant][graphmode][fx][fp16] Add fp16 support for tanh
* **#52863 [quant][graphmode][fx][fp16] Add fp16 support for sigmoid**
* #52862 [quant][graphmode][fx][fp16] Add fp16 support for layer_norm
* #52811 [quant][graphmode][fx][fp16] Add fp16 support for torch.sum
* #52810 [quant][graphmode][fx][fp16] Add fp16 support for div
* #52809 [quant][graphmode][fx][fp16] Add fp16 support for sub

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

jerryzh168 committed:
[quant][graphmode][fx][fp16] Add fp16 support for sigmoid

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]

vkuzo commented on torch/quantization/fx/quantization_patterns.py:
```diff
@@ -841,7 +841,12 @@ class FixedQParamsOpQuantizeHandler(QuantizeHandler):
+        qconfig = quantizer.qconfig_map[node.name]
+        dtypes = get_qconfig_dtypes(qconfig)
+        if dtypes == (torch.float16, torch.float16, None):
+            return quantizer.quantized_graph.node_copy(node, load_arg(quantized=False))
+        else:
+            return quantizer.quantized_graph.node_copy(node, load_arg(quantized=None))
```
just for my own understanding, what's the difference between `load_arg(quantized=False)` and `load_arg(quantized=None)`?

jerryzh168 replied:
load_arg(quantized=False) means dequantize the node if needed, load_arg(quantized=None) means do nothing, just copy the node

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fp16] Add fp16 support for sigmoid"


Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_fixed_qparams_ops_fp16

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26672273](https://our.internmc.facebook.com/intern/diff/D26672273)

[ghstack-poisoned]
